src/test/java/com/walmartlabs/x12/asn856/DefaultAsn856ParserTest.java

```diff
@@ -34,12 +40,28 @@
     public void init() {
         asnParser = new DefaultAsn856Parser();
     }
+
+    @Test
+    public void test_Parsing_SourceIsNull() throws IOException {
+        String sourceData = null;
+        X12Document x12 = asnParser.parse(sourceData);
+        assertNull(x12);
+    }
+
+    @Test
+    public void test_Parsing_SourceIsEmpty() throws IOException {
+        String sourceData = "";
+        X12Document x12 = asnParser.parse(sourceData);
+        assertNull(x12);
+    }
+
     @Test
-    public void testParsingShipmentWithMissingDxe() throws IOException {
+    public void test_Parsing_Asn856() throws IOException {
         byte[] asnBytes = Files.readAllBytes(Paths.get("src/test/resources/asn856/asn856.txt"));
-        Asn856 asn = asnParser.parse(new String(asnBytes));
+        X12Document x12 = asnParser.parse(new String(asnBytes));
```

Comment:
`StandardX12Document asn = asnParser.parse(new String(asnBytes));` would work because `DefaultAsn856Parser extends AbstractStandardX12Parser<StandardX12Document>`